OpenRA.Mods.Common/Traits/Render/Hovers.cs

```diff
@@ -13,41 +13,79 @@
 using System.Collections.Generic;
 using System.Drawing;
 using System.Linq;
+using OpenRA;
 using OpenRA.Graphics;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Changes the visual Z position periodically.")]
-	public class HoversInfo : ConditionalTraitInfo, Requires<IMoveInfo>
+	public class HoversInfo : PausableConditionalTraitInfo, Requires<IMoveInfo>
 	{
 		[Desc("Amount of Z axis changes in world units.")]
 		public readonly int OffsetModifier = -43;
 
 		public readonly int MinHoveringAltitude = 0;
 
+		[Desc("Amount of ticks it takes to reach OffsetModifier.")]
+		public readonly int Ticks = 25;
+
+		[Desc("Amount of ticks it takes to fall to the ground from the highest point when disabled.")]
+		public readonly int FallTicks = 12;
+
+		[Desc("Amount of ticks it takes to rise from the ground to InitialHeight.")]
+		public readonly int RiseTicks = 17;
+
+		public readonly int InitialHeight = 384;
+
 		public override object Create(ActorInitializer init) { return new Hovers(this, init.Self); }
 	}
 
-	public class Hovers : ConditionalTrait<HoversInfo>, IRenderModifier
+	public class Hovers : PausableConditionalTrait<HoversInfo>, IRenderModifier, ITick
```

Comment:
Both places that check disabled/paused check both, so in practice there's no difference in behavior. In my opinion just leave the trait as conditional instead of pausable-conditional, as I can't really come up with an idea to differentiate between the two, either.

Reply:
sounds fair, will do